workers/unity/Assets/Playground/Scripts/Worker/AbstractMobileClientWorker.cs

```diff
@@ -0,0 +1,81 @@
+﻿using Improbable.Gdk.Core;
+using Improbable.Gdk.Mobile;
+using Playground;
+using UnityEngine;
+using UnityEngine.UI;
+
+public abstract class AbstractMobileClientWorker : MobileWorkerConnector
+{
+    public GameObject ConnectionPanel;
+    public GameObject Level;
+
+    private GameObject levelInstance;
+    private bool connected;
+    private InputField input;
+    private Button button;
+    private Text errorMessage;
+
+    public void Awake()
+    {
+        input = ConnectionPanel.transform.Find("ConnectInput").GetComponent<InputField>();
+        button = ConnectionPanel.transform.Find("ConnectButton").GetComponent<Button>();
+        errorMessage = ConnectionPanel.transform.Find("ConnectionError").GetComponent<Text>();
+
+        input.text = PlayerPrefs.GetString("cachedIp");
+    }
+
+    public void Start()
+    {
+        button.onClick.AddListener(Connect);
+    }
+
+    public async void Connect()
+    {
+        errorMessage.text = "";
+        await Connect(WorkerUtils.UnityClient, new ForwardingDispatcher()).ConfigureAwait(false);
+    }
+
+    protected override void HandleWorkerConnectionEstablished()
+    {
+        WorkerUtils.AddClientSystems(Worker.World);
```

Comment:
Are we sure these should be the same systems?

Reply:
I thought the controls one at least was different.

Reply:
Controls are not in these PRs. It's an MVP of connecting and being able to dev locally; controls implementation seemed too convoluted and time consuming for the time being.